pkg/pillar/cmd/zedrouter/flowstats.go

```diff
@@ -179,36 +190,55 @@ func FlowStatsCollect(ctx *zedrouterContext) {
 			// temp print out the flow "tuple" and stats per app/bridge
 			for i, tuple := range timeOutTuples { // search for flowstats by bridge
 				var aclattr aclAttr
+				var bridgeName, aclaction string
+				var aclNum int
 
 				appN := tuple.appNum
 				if int(appN) != appIdx { // allow non-App flows to be uploaded
 					//log.Infof("FlowStats: appN %d, appIdx %d not match", appN, appIdx)
 					continue
 				}
 
-				tmpMap := instData.ipaclattr[int(appN)]
-				if tmpMap != nil {
-					if _, ok := tmpMap[int(tuple.aclNum)]; !ok {
-						log.Infof("FlowStats: == can not get acl map with aclN, should not happen appN %d, aclN %d; %s\n",
+				if tuple.aclNum != DropMarkValue {
+					tmpMap := instData.ipaclattr[int(appN)]
+					if tmpMap != nil {
+						if _, ok := tmpMap[int(tuple.aclNum)]; !ok {
+							log.Infof("FlowStats: == can not get acl map with aclN, should not happen appN %d, aclN %d; %s\n",
+								appN, tuple.aclNum, tuple.String())
+							continue
+						}
+						aclattr = tmpMap[int(tuple.aclNum)]
+					} else {
+						log.Infof("FlowStats: == can't get acl map with appN, should not happen, appN %d, aclN %d; %s\n",
 							appN, tuple.aclNum, tuple.String())
 						continue
 					}
-					aclattr = tmpMap[int(tuple.aclNum)]
-				} else {
-					log.Infof("FlowStats: == can't get acl map with appN, should not happen, appN %d, aclN %d; %s\n",
-						appN, tuple.aclNum, tuple.String())
-					continue
-				}
-				if aclattr.aclNum == 0 {
-					log.Infof("FlowStats: == aclN zero in attr, appN %d, aclN %d; %s\n", appN, tuple.aclNum, tuple.String())
-					// some debug info
-					continue
-				}
-				bridgeName := aclattr.bridge
-				if strings.Compare(bnx, bridgeName) != 0 {
-					log.Infof("FlowStats: == bridge name not match %s, %s\n", bnx, bridgeName)
-					continue
+					if aclattr.aclNum == 0 {
+						log.Infof("FlowStats: == aclN zero in attr, appN %d, aclN %d; %s\n", appN, tuple.aclNum, tuple.String())
+						// some debug info
+						continue
+					}
+
+					bridgeName = aclattr.bridge
+					if strings.Compare(bnx, bridgeName) != 0 {
+						log.Infof("FlowStats: == bridge name not match %s, %s\n", bnx, bridgeName)
+						continue
+					}
+					scope.Intf = aclattr.intfname // App side DomU internal interface name
+					aclaction = aclattr.action
+					aclNum = int(aclattr.aclNum)
+				} else { // conntrack mark aclNum field being 0xffffff
```

Comment:
For the drop mark rules MSB of marking will have the App num. It's not clear to me when this code will be needed.

Reply:
It has the app-ID, but it does not have the bridge-name. each pack of protobuf is app/bridge. thus we need to find out which bridge this dropped flow belongs to.